Production/src/Selectors.cc

```diff
@@ -15,12 +15,47 @@ std::shared_ptr<TauJetSelector> TauJetSelector::Make(const std::string& name)
     throw analysis::exception("Unknown selector name = '%1%'") % name;
 }
 
+bool muonveto (const std::vector<pat::Muon>& muons, const pat::Muon *ref_muon, const reco::Vertex& primaryVertex){
+    for(const pat::Muon& muon : muons) {
+        if(muon.pt() > 10 && std::abs(muon.eta()) < 2.4 && muon.isMediumMuon() && PFRelIsolation(muon) < 0.30 && std::abs(muon.muonBestTrack()->dxy(primaryVertex.position())) < 0.2
+                && std::abs(muon.muonBestTrack()->dz(primaryVertex.position())) < 0.0045&& &muon != ref_muon){
+            return true;
+        }
+    }
+    return false;
+}
+
+bool electronveto (const std::vector<pat::Electron>& electrons, const reco::Vertex& primaryVertex, const float rho){
+    for(const pat::Electron& electron : electrons) {
+        if(electron.pt() > 10 && std::abs(electron.eta()) < 2.5 && electron.electronID("mvaEleID-Fall17-noIso-V2-wp90") > 0.5f && PFRelIsolation_e(electron, rho)<0.3){
+            return true;
+        }
+    }
+    return false;
+}
+
+bool dimuonveto (const std::vector<pat::Muon>& muons, const pat::Muon *ref_muon, const reco::Vertex& primaryVertex){
+    std::vector<pat::Muon> dimuon_candidates; // vector of all muons that pass selection
```

Comment:
- should be `std::vector<const pat::Muon*>`

Reply:
in both lines 37 and 38?